kythe/go/extractors/config/preprocessor/README.md

```diff
@@ -0,0 +1,73 @@
+# Build Config Preprocessing
+
+A number of builders require custom modification to a repo's build config before
+Kythe extraction will work correctly.
+
+## Preprocessor
+
+This is the top level binary for doing preprocessing.  The default operation is
+simply `./preprocess <build-file>`, which will detect what format the build file
+is in and perform necessary steps.
+
+## Invoking via docker / Cloud Build
+
+TODO(#3158): Actually fill this out with a Dockerfile and example commands.
+
+## Specific Builders
+
+### Gradle
+
+In order to run Kythe extraction on a gradle repo, we must first modify the
+`gradle.build` file to hook into a separate javac wrapper binary.
+`gradle_build_modifier.go` takes an input `gradle.build` file and appends the
+bits necessary for replacing javac calls with Kythe's `javac-wrapper.sh`.
+
+```
+allprojects {
+  gradle.projectsEvaluated {
+    tasks.withType(JavaCompile) {
+      options.fork = true
+      options.forkOptions.executable = '/opt/kythe/extractors/javac-wrapper.sh'
+    }
+  }
+}
+```
+
+If the input file already contains reference to
+`options.forkOptions.executable`, then `gradle_build_modifier.go` does nothing.
```

Comment:
Can we not extract from those builds?

Reply:
That is a good question.

I have been banking on the theory that

1. yes we will be able to extract from those builds, and
2. we can do this by just overriding the executable, and then pointing to the overridden executable in javac-wrapper.sh

Then we'd just need some small modification in here in the mutator, and also a way to propagate the overridden bit back out and stuff it into REAL_JAVAC or whatever environment variable (I forget off the top of my head).

Of course, I have no proof that any of this would work.

In any event, filing an issue:

#3162